orttraining/orttraining/test/python/orttraining_test_python_bindings.py

```diff
@@ -0,0 +1,243 @@
+    output_names = []
+    for i, model_input in enumerate(model_inputs):
+        input_name = f"input-{i}"
+        input_names.append(input_name)
+        dynamic_axes[input_name] = {}
+        for dim_idx in range(len(model_input.shape)):
+            dynamic_axes[input_name].update({dim_idx: f"{input_name}_dim{dim_idx}"})
+
+    for i, model_output in enumerate(model_outputs):
+        output_name = f"output-{i}"
+        output_names.append(output_name)
+        dynamic_axes[output_name] = {}
+        for dim_idx in range(len(model_output.shape)):
+            dynamic_axes[output_name].update({dim_idx: f"{output_name}_dim{dim_idx}"})
+
+    f = io.BytesIO()
+    torch.onnx.export(
+        torch_model,
+        model_inputs,
+        f,
+        input_names=input_names,
+        output_names=output_names,
+        opset_version=14,
+        do_constant_folding=False,
+        training=torch.onnx.TrainingMode.TRAINING,
+        dynamic_axes=dynamic_axes,
+        export_params=True,
+        keep_initializers_as_inputs=False,
+    )
+    return onnx.load_model_from_string(f.getvalue())
+
+
+def _create_training_models():
+    # Given
+    device = "cuda"
+    batch_size, input_size, hidden_size, output_size = 64, 784, 500, 10
+    _, onnx_model = _get_models(device, batch_size, input_size, hidden_size, output_size)
+
+    # Build the onnx model with loss
+    simple_model = SimpleModelWithCrossEntropyLoss()
+    with onnxblock.onnx_model(onnx_model) as accessor:
+        _ = simple_model(onnx_model.graph.output[0].name)
+        eval_model = accessor.eval_model
+
+    optimizer = onnxblock.optim.AdamW()
+    with onnxblock.onnx_model() as accessor:
+        _ = optimizer(simple_model.parameters())
+        optimizer_model = accessor.model
+
+    return simple_model, onnx_model, optimizer_model, eval_model
+
+
+def test_train_step():
+    # Initialize Models
+    simple_model, onnx_model, _, _ = _create_training_models()
+    trainable_params, non_trainable_params = simple_model.parameters()
+
+    # Generating random data for testing.
+    inputs = torch.randn(64, 784).numpy()
+    labels = torch.randint(high=10, size=(64,), dtype=torch.int32).numpy()
+    forward_inputs = OrtValueVector()
+    forward_inputs.reserve(2)
+    forward_inputs.push_back(OrtValue.ortvalue_from_numpy(inputs)._ortvalue)
+    forward_inputs.push_back(OrtValue.ortvalue_from_numpy(labels)._ortvalue)
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Save models & checkpoint files to load them later.
+        checkpoint_file_path = os.path.join(temp_dir, "checkpoint")
+        onnxblock.save_checkpoint((trainable_params, non_trainable_params), checkpoint_file_path)
+
+        model_file_path = os.path.join(temp_dir, "training_model.onnx")
+        onnx.save(onnx_model, model_file_path)
+
+        # Create a Module.
+        model = Module(model_file_path, checkpoint_file_path)
+        model.train()
+        fetches = model(forward_inputs)
```

Comment:
since you have the pytorch model can you compare the outputs?

Reply:
to compare that, we have to expose the Parameter Object first, this will be included in the next PR.

Reply:
Sorry, I may have missed this. Why do you need the Parameters to compare the output values?

Reply:
so we can pull the parameters object and compare before and after to make sure they got changed after running the train, and optimizer step? 
or do you want to compare just the loss to make sure it got improved after a train, optimize step?

Reply:
The outputs of the `TrainStep` include only the forward outputs. In this case, `fetches` would be the forward outputs. And should be the same as pytorch's forward outputs. And therefore, can be compared.